<launch>
  <!-- ===== 공통 인자 ===== -->
  <arg name="ref_file"       default="/root/ws/src/roscpp_morai/map/ref.txt"/>
  <arg name="path_file"      default="/root/ws/src/roscpp_morai/map/Path.txt"/>
  <arg name="frame_id"       default="map"/>
  <arg name="highlight_dist" default="30.0"/>
  <arg name="line_width"     default="0.2"/>

  <!-- Pure Pursuit 파라미터 -->
  <arg name="wheelbase"      default="3.0"/>
  <arg name="lookahead"      default="20.0"/>
  <!-- 속도 5 km/h 고정 (노드에서 longlCmdType=2 로 velocity 전송) -->
  <arg name="target_kmh"     default="5.0"/>

  <!-- RViz 사용 여부/설정 -->
  <arg name="use_rviz"       default="true"/>
  <!-- 저장해둔 RViz 설정 파일이 있으면 넣고, 없으면 비워두세요 -->
  <arg name="rviz_config"    default=""/>

  <!-- ===== Path 시각화 노드 ===== -->
  <node pkg="roscpp_morai" type="path_visualizer" name="path_visualizer" output="screen">
    <param name="ref_file"       value="$(arg ref_file)"/>
    <param name="path_file"      value="$(arg path_file)"/>
    <param name="frame_id"       value="$(arg frame_id)"/>
    <param name="highlight_dist" value="$(arg highlight_dist)"/>
    <param name="line_width"     value="$(arg line_width)"/>
    <!-- 현재 위치 노랑 점 표시 -->
    <param name="show_current"   value="true"/>
  </node>

  <!-- ===== Pure Pursuit (5 km/h 고정속) 노드 ===== -->
  <!-- <node pkg="roscpp_morai" type="pp_fixed_speed" name="pp_fixed_speed" output="screen">
    <param name="ref_file"   value="$(arg ref_file)"/>
    <param name="path_file"  value="$(arg path_file)"/>
    <param name="wheelbase"  value="$(arg wheelbase)"/>
    <param name="lookahead"  value="$(arg lookahead)"/>
    <param name="target_kmh" value="$(arg target_kmh)"/>  <!-- 기본 5 -->
  </node> -->

  <!-- ===== RViz ===== -->
  <!-- rviz_config 인자가 비어있지 않으면 지정된 설정으로, 비어있으면 기본으로 실행 -->
  <group if="$(arg use_rviz)">
    <node pkg="rviz" type="rviz" name="rviz" output="screen" if="$(eval rviz_config != '')"
          args="-d $(arg rviz_config)"/>
    <node pkg="rviz" type="rviz" name="rviz" output="screen" unless="$(eval rviz_config != '')"/>
  </group>
</launch>
